src/System.Management.Automation/engine/CommandCompletion/PseudoParameterBinder.cs

```diff
@@ -1862,13 +1862,19 @@ private static bool IsTypeEquivalent(Type argType, Type paramType)
             while (unboundArgumentsIndex < unboundArgumentsCollection.Count)
             {
                 AstParameterArgumentPair argument = unboundArgumentsCollection[unboundArgumentsIndex++];
-                if (!argument.ParameterSpecified)
+                bool splatted = argument is AstPair astPair
+                    && astPair.Argument is VariableExpressionAst argumentVariable
+                    && argumentVariable.Splatted;
+                if (!splatted && !argument.ParameterSpecified)
```

Comment:
I think this could be simplified to:
```csharp
if (splatted)
{
    continue;
}
```
Then the rest of the loop can go back to the code as it was before.